elastic/apm-server
# modeldecoder/v2: add `metricset.internal`

## Motivation/summary

Introduce `internal` field to metricsets, enabling agents to indicate that a metricset contains only internal (non-application) metrics, such as runtime, system, and process metrics.

If `internal` is set to `true`, then the server will discard any metrics in the metricset that it does not know to be internal. This is necessary as the internal metrics data stream uses strict mapping.

If `internal` is is not set at all, then the server will check if all metrics are known to by internal; if any are not, then the entire set is sent to a service-specific application metrics data stream.

## Checklist




For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

1. Run all opbeans services and ensure no service-specific data streams are created.
2. We should also modify one of the agents to set the `internal` flag for internal metrics, and check that known internal metrics are stored in the internal metrics data stream, and all others are discarded

## Related issues

https://github.com/elastic/apm-server/issues/7520
